cloudbridge/cloud/providers/gce/services.py

```diff
@@ -797,14 +812,27 @@ def create(self, network, cidr_block, name=None, zone=None):
                 return None
             subnets = self.list(network, region)
             for subnet in subnets:
-                cb.log.warning('subnet ID: %s', subnet.id)
                 if subnet.id == response['targetId']:
                     return subnet
         except:
             return None
 
     def get_or_create_default(self, zone=None):
-        raise NotImplementedError('To be implemented')
+        """
+        Every GCP project comes with a default auto mode VPC network. An auto
+        mode VPC network has exactly one subnetwork per region. This method
+        returns the subnetwork of the default network that spans the given
+        zone.
+        """
+        network = self.provider.network.get_or_create_default()
+        subnets = self.list(network, zone)
+        if len(subnets) > 1:
+            cb.log.warning('The default network has more than one subnetwork '
```

Comment:
This seems to go against the intent of this method. As a user, by calling `get_or_create_default` I'd expect to get something back. I guess it's not likely that this condition will be triggered as it's likely there will be only one subnet for the default network so it's probably ok but I wonder if perhaps we should instead just return a random  subnet here as well?

Reply:
Note that this line just logs a warning. Next lines will actually return something.

Reply:
Ahh, yes, I see. I just took the next `if` as `else if`...